addons/l10n_es_edi_facturae/models/account_edi_format.py

```diff
@@ -0,0 +1,465 @@
+            return {
+                'post': self._l10n_es_edi_facturae_post_invoices,
+                'edi_content': self._l10n_es_edi_facturae_xml_edi_content,
+            }
+        return {}
+
+    def _needs_web_services(self):
+        # EXTENDS account_edi
+        return False if self.code == 'es_face' else super()._needs_web_services()
+
+    #############################
+    #   EDI SPECIFIC METHODS    #
+    #############################
+
+    def _l10n_es_edi_facturae_xml_edi_content(self, invoice):
+        return self.l10n_es_edi_facturae_export_facturae(invoice).encode()
+
+    def _l10n_es_edi_facturae_post_invoices(self, invoices):
+        # Ensure a certificate is available.
+        certificate = self.env['l10n_es_edi_facturae.certificate'].search([("company_id", 'in', invoices.company_id.ids)])
+        if not certificate:
+            return {inv: {
+                'error': _("Please configure the certificate for Facturae."),
+                'blocking_level': 'error',
+            } for inv in invoices}
+
+        # Generate the XMLs.
+        res = {}
+        for inv in invoices:
+            xml_name = f'{inv.name.replace("/", "_")}_facturae_signed.xml'
+            attachment = self.env['ir.attachment'].create({
+                'name': xml_name,
+                'raw': self.l10n_es_edi_facturae_render_facturae(inv),
+                'res_model': 'account.move',
+                'res_id': inv.id,
+                'mimetype': 'application/xml'
+            })
+            res[inv] = {'success': True, 'attachment': attachment}
+        return res
+
+    @api.model
+    def l10n_es_edi_facturae_inv_lines_to_items(self, invoice, conv_rate, needs_equivalency, fmt_curr):
+        """
+        Convert the invoice lines to a list of items required for the Facturae xml generation
+
+        :param account.move invoice:    The invoice containing the invoice lines
+        :param float conv_rate:         The conversion rate from the invoice currency to the company currency
+        :param bool needs_equivalency:  True if the invoice uses multi-currencies
+        :param func fmt_curr:         The formatting function to use for the amount in the invoice currency
+        :return tuple:                  A tuple containing the Face items, the taxes and the invoice totals data.
+        """
+        items = []
+        totals = {
+            'total_gross_amount': 0.,
+            'total_general_discounts': 0.,
+            'total_general_surcharges': 0.,
+            'total_taxes_withheld': 0.,
+            'total_payments_on_account': 0.,
+            'amounts_withheld': 0.,
+        }
+        taxes = []
+        inv_curr = invoice.currency_id
+        eur_curr = invoice.company_currency_id
+        for line in invoice.invoice_line_ids:
+            invoice_line_values = EDI_XML_DICT_TEMPLATES['invoice_line'].copy()
+
+            total_price = line.quantity * line.price_unit * conv_rate  # This isn't a mistake
```

Comment:
In case the tax is included or excluded this will behave differently, so why is this not a mistake?